plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/storage/TestTrinoHudiStorage.java

```diff
@@ -0,0 +1,294 @@
+ */
+package io.trino.plugin.hudi.storage;
+
+import io.trino.filesystem.TrinoFileSystem;
+import io.trino.filesystem.local.LocalFileSystemFactory;
+import io.trino.spi.block.TestingSession;
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.io.storage.TestHoodieStorageBase;
+import org.apache.hudi.io.util.IOUtils;
+import org.apache.hudi.storage.HoodieStorage;
+import org.apache.hudi.storage.StoragePath;
+import org.apache.hudi.storage.StoragePathInfo;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestTrinoHudiStorage
+        extends TestHoodieStorageBase
+{
+    private static final byte[] EMPTY_BYTES = new byte[] {};
+
+    private static TrinoFileSystem fileSystem;
+
+    @BeforeAll
+    static void setUp()
+    {
+        fileSystem = new LocalFileSystemFactory(Path.of("/tmp"))
+                .create(TestingSession.SESSION);
+    }
+
+    @Override
+    protected HoodieStorage getStorage(Object fileSystem, Object config)
+    {
+        return new TrinoHudiStorage((TrinoFileSystem) fileSystem, (TrinoStorageConfiguration) config);
+    }
+
+    @Override
+    protected TrinoFileSystem getFileSystem(Object config)
+    {
+        return fileSystem;
+    }
+
+    @Override
+    protected TrinoStorageConfiguration getConf()
+    {
+        return new TrinoStorageConfiguration();
+    }
+
+    @Override
+    protected String getTempDir()
+    {
+        return "local:" + this.tempDir.toUri().getPath();
+    }
+
+    @Override
+    @Test
+    public void testGetUri()
+            throws URISyntaxException
+    {
+        assertThat(getStorage().getUri()).isEqualTo(new URI(""));
+    }
+
+    @Override
+    @Test
+    public void testListing()
+            throws IOException
+    {
+        HoodieStorage storage = getStorage();
+        // Full list:
+        // w/1.file
+        // w/2.file
+        // x/1.file
+        // x/2.file
+        // x/y/1.file
+        // x/y/2.file
+        // x/z/1.file
+        // x/z/2.file
+        prepareFilesOnStorage(storage);
+
+        validatePathInfoList(
+                Arrays.stream(new StoragePathInfo[] {
+                        getStoragePathInfo("x/1.file", false),
+                        getStoragePathInfo("x/2.file", false),
+                        getStoragePathInfo("x/y", true),
+                        getStoragePathInfo("x/z", true)
+                }).collect(Collectors.toList()),
```

Comment:
Replace with ImmutableList builder. Same for others.

Reply:
Fixed.